dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/https/HttpsURLConnectionFactory.java

```diff
@@ -0,0 +1,292 @@
+            LOG.fine("SSL socketFactory is NULL"); // Liberty Change
+            if (tlsClientParameters.getSslContext() != null) {
+                // Use the SSLContext which was set
+                LOG.fine("Get SSLContext from tlsClientParameters"); // Liberty Change
+                ctx = tlsClientParameters.getSslContext();
+            } else {
+		// Liberty Change Start
+	        if (tlsClientParameters.isJaxrsClient() || (tlsClientParameters.getTrustManagers() != null && 
+			tlsClientParameters.getKeyManagers() != null)) {
+                   // Get sslcontext with tlsClientParameters's Trust Managers, Key Managers, etc
+                   LOG.fine("Get SSLContext with tlsClientParameters's Trust Managers, Key Managers");
+                   ctx = org.apache.cxf.transport.https.SSLUtils.getSSLContext(tlsClientParameters);
+	        }
+		else {
+                   LOG.fine("No trustManagers set on tlsClientParameters, so use Liberty's DefaultSSLSocketFactory");
+                   socketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
+		}
+		// Liberty Change End
+            }
+	    if (ctx != null) { // do we have SSL context?
+                LOG.fine("Create new socketFactory from SSLContext and set Ciphersuites");
+                String[] cipherSuites =
+                    SSLUtils.getCiphersuitesToInclude(tlsClientParameters.getCipherSuites(),
+                                                  tlsClientParameters.getCipherSuitesFilter(),
+                                                  ctx.getSocketFactory().getDefaultCipherSuites(),
+                                                  SSLUtils.getSupportedCipherSuites(ctx),
+                                                  LOG);
+                // The SSLSocketFactoryWrapper enables certain cipher suites from the policy.
+                String protocol = tlsClientParameters.getSecureSocketProtocol() != null ? tlsClientParameters
+                                            .getSecureSocketProtocol() : ctx.getProtocol();
+                socketFactory = new SSLSocketFactoryWrapper(ctx.getSocketFactory(), cipherSuites, protocol);
+	    }
+
+	    if (socketFactory != null) {
+               LOG.fine("SSL socketFactory: " +  socketFactory.getClass().getCanonicalName()); // Liberty Change
+	    }
+            //recalc the hashcode since some of the above MAY have changed the tlsClientParameters
+            lastTlsHash = tlsClientParameters.hashCode();
+        } else {
+            LOG.fine("SSL socketFactory already initialized!");  // Liberty Change
+           // ssl socket factory already initialized, reuse it to benefit of keep alive
+        }
+
+        HostnameVerifier verifier = org.apache.cxf.transport.https.SSLUtils
+            .getHostnameVerifier(tlsClientParameters);
+
+	// Liberty Change Start
+	if (verifier != null) {
+           LOG.fine("Hostname verifier obtained from SSLUtils.getHostnameVerifier: " +  verifier.getClass().getCanonicalName());
```

Comment:
Suggestion:  Remove the if check and change the line to the following so you'll know if it is null or not more easily:
`            LOG.fine(""Hostname verifier obtained from SSLUtils.getHostnameVerifier: " + (verifier==null? "null": verifier.getClass().getCanonicalName()));
`

Reply:
I think we should leave the if check, but only print a message if the verifier is null.